core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -1461,6 +1461,11 @@ package object config {
       .doubleConf
       .createWithDefault(0.75)
 
+  private[spark] val SPECULATION_TASK_DURATION_THRESHOLD =
+    ConfigBuilder("spark.speculation.task.duration.threshold")
+      .timeConf(TimeUnit.MILLISECONDS)
+      .createOptional
```

Comment:
can we add a simple `.doc` to explain what this config does?

Reply:
Do I need to change https://github.com/apache/spark/blob/master/docs/configuration.md?

Reply:
yes please